app/controllers/admin_scope/company_registrations_controller.rb

```diff
@@ -1,18 +1,11 @@
 module AdminScope
   class CompanyRegistrationsController < BaseController
     expose :company, find_by: :slug
-    expose :company_policy, -> { set_company_policy }
 
     def update
-      redirect_to company_path company and return unless company_policy.become_to_author?
+      redirect_to company_path company and return unless policy(company).become_to_author?
```

Comment:
I think we can use `rescue_from Pundit::NotAuthorizedError` construction in `BaseController` like as:

```ruby
# app/concerns/authorizable.rb

module Authorizable
   included do
    include Pundit

    rescue_from Pundit::NotAuthorizedError, with: :profile_not_authorized
  end
end

# app/controllers/admin_scope/company_registrations_controller.rb
...
def update
  authorize :company, :become_to_author?

  company.authors.push(current_user)
  
  redirect_back fallback_location: companies_path, notice: "Now you became an author"  
end
```

Reply:
I don`t understand why we neet Authorizable?

Reply:
I saw code and I noticed that I had redirect to root_path in case of NotAuthorizedError because rails-base already has module Authorization: app/controllers/concerns/authorization
But I read article in documentation and now I know how it works